source/common/conn_pool/conn_pool_base.h

```diff
@@ -23,6 +23,7 @@ class ConnPoolImplBase;
 struct AttachContext {
   // Add a virtual destructor to allow for the dynamic_cast ASSERT in typedContext.
   virtual ~AttachContext() = default;
+  uint64_t connection_id_;
```

Comment:
Adding connection id here means that it would need to propagate through ``HttpAttachContext`` or ``TcpAttachContext``, when calling ``newStream``, so a change to ``newStream`` signature is required with a new paramater:
https://github.com/envoyproxy/envoy/blob/a9571872dbf1c3c125ab0aaba086e5416e1048f9/envoy/http/conn_pool.h#L114

Alternatively, ConnPoolImplBase could be change so it also gets the connection id in the ctor:
https://github.com/envoyproxy/envoy/blob/a9571872dbf1c3c125ab0aaba086e5416e1048f9/source/common/conn_pool/conn_pool_base.h#L186

Will appreciate guidance if any of these approaches makes sense